provisioning/baremetal_pod.go

```diff
@@ -109,11 +109,20 @@ func buildEnvVar(name string, baremetalProvisioningConfig *metal3iov1alpha1.Prov
 			Name:  name,
 			Value: *value,
 		}
-	} else {
+	} else if name == provisioningIP && baremetalProvisioningConfig.ProvisioningNetwork == metal3iov1alpha1.ProvisioningNetworkDisabled {
```

Comment:
This logic can be added to https://github.com/openshift/cluster-baremetal-operator/blob/master/provisioning/baremetal_config.go#L131. The get*() methods in baremetal_config.go are meant to take care of all the interdependencies between the config. Implementing it like this will spread the knowledge of config interdependencies in multiple places.

Reply:
That function returns a string pointer. I need to construct a different kind of environment variable using a FieldRef.

Reply:
Is the status.hostIP going to be localhost?

Reply:
I agree it would be nicer if this logic wasn't here, but I didn't find an easy way to do it. I'm happy to refactor it later if we can let this land. There's still a lot left to do for 3704.

Reply:
So, should this value be an actual IP address vs "localhost"?

Reply:
It is Ok to merge even with some outstanding comments. We can capture it here and fix it later.

Reply:
It's filled in by the runtime, but yes it will be an actual IP address

Reply:
> So, should this value be an actual IP address vs "localhost"?

The ironic containers themselves do get the host IP.  The FieldRef in this code is telling kubernetes to wire that value in at run time. CBO can't know the value since it won't know where the pod will land.